src/server/scripts/Outland/BlackTemple/boss_shade_of_akama.cpp

```diff
@@ -25,251 +25,175 @@ enum Says
-
-    EVENT_SHADE_CHECK_DISTANCE      = 10,
-    EVENT_SHADE_RESET_ENCOUNTER     = 11,
-    EVENT_SHADE_GATHER_NPCS         = 12,
-
-    EVENT_SUMMON_WAVE_B             = 20,
-    EVENT_SUMMON_ASHTONGUE_SORCERER = 21,
-    EVENT_SUMMON_ASHTONGUE_DEFENDER = 22,
-
-    EVENT_AKAMA_SCENE0              = 29,
-    EVENT_AKAMA_SCENE1              = 30,
-    EVENT_AKAMA_SCENE2              = 31,
-    EVENT_AKAMA_SCENE3              = 32,
-    EVENT_AKAMA_SCENE4              = 33,
-    EVENT_AKAMA_SCENE5              = 34,
-    EVENT_AKAMA_SCENE6              = 35,
-    EVENT_AKAMA_SCENE7              = 36
+    POINT_ENGAGE                    = 0,
+    POINT_OUTRO                     = 1,
+
+    ACTION_GENERATOR_START          = 1,
+    ACTION_GENERATOR_STOP           = 2,
+    ACTION_GENERATOR_DESPAWN_ALL    = 3,
+
+    COUNTER_SPAWNS_MAX              = 20,   // Max number of spawns for each generator, number chosen at random
+
+    ACTION_AKAMA_START_OUTRO        = 1,
+
+    FACTION_DEFAULT                 = 1820,
+    FACTION_ENGAGE                  = 1868
 };
 
+Position AkamaEngage = { 517.4877f, 400.79926f, 112.77704f };
+Position AkamaOutro = { 469.0867f,  401.0793f,  118.52704f, 0.087266460061073303f };
+Position ShadeEngage = { 512.48773f, 400.8283f, 112.77704f };
+
 struct boss_shade_of_akama : public BossAI
 {
-    boss_shade_of_akama(Creature* creature) : BossAI(creature, DATA_SHADE_OF_AKAMA), summonsChanneler(me), summonsGenerator(me)
-    {
-        events2.ScheduleEvent(EVENT_SHADE_GATHER_NPCS, 1000);
-    }
-
-    SummonList summonsChanneler;
-    SummonList summonsGenerator;
-    EventMap events2;
+    boss_shade_of_akama(Creature* creature) : BossAI(creature, DATA_SHADE_OF_AKAMA) { }
 
-    void ChannelersAction(int32 action)
-    {
-        for (SummonList::const_iterator i = summonsChanneler.begin(); i != summonsChanneler.end(); ++i)
-            if (Creature* summon = ObjectAccessor::GetCreature(*me, *i))
-            {
-                if (action == ACTION_CHANNELERS_START_CHANNEL)
-                {
-                    summon->CastSpell(me, SPELL_SHADE_SOUL_CHANNEL, true);
-                    summon->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
-                }
-                else if (action == ACTION_START_ENCOUNTER)
-                {
-                    summon->RemoveUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
-                }
-                else if (action == ACTION_KILL_CHANNELERS)
-                {
-                    Unit::Kill(me, summon);
-                }
-            }
-    }
+    std::list<Creature*> channelers;
+    std::list<Creature*> generators;
 
     void Reset() override
     {
-        BossAI::Reset();
-        me->SetReactState(REACT_PASSIVE);
         me->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
-        me->SetImmuneToAll(true);
         me->SetWalk(true);
+        me->SetReactState(REACT_DEFENSIVE);
+        BossAI::Reset();
     }
 
     void EnterEvadeMode(EvadeReason why) override
     {
```

Comment:
me->GetCreatureListWithEntryInGrid(channelers, NPC_ASHTONGUE_CHANNELER, 40.0f);
            for (auto const& channeler : channelers)
                channeler->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);

Reply:
Maybe we can make them minions and then you can do for all minions. But tbh idk how this fight is and if they're suitable as minions. See Magtheridon for examples.

Reply:
Will add it to OnEvade instead